lib/http.js

```diff
@@ -0,0 +1,181 @@
+    }
+  };
+
+  // TODO: maybe skip revalidation for our use case and just don't update
+  // 5 minute time out? ... or something! ... we don't want to do a 304
+  // all the time for lots of different clients, only every 5 minutes or so
+
+  if(policy && policy.satisfiesWithoutRevalidation(request)) {
+    // cached response is reusable, send it
+    response.headers = policy.responseHeaders();
+    return response;
+  }
+
+  if(policy) {
+    // ask origin server if cached response can be used
+    request.headers = policy.revalidatedPolicy(request, response);
+  }
+
+  // response not cached yet or not reusable, fetch again
+  const newResponse = await getManifest({manifestUrl, request});
+
+  if(policy) {
+    // update policy
+    const result = policy.revalidatedPolicy(request, response);
+    policy = result.policy;
+    response = result.modified ? newResponse : response;
+  } else {
+    // create policy
+    response = newResponse;
+    policy = new CachePolicy(request, response);
+  }
+
+  // even if HTTP cache headers say not to cache, we cache for a short time
+  const ttl = policy.storable() ?
+    policy.timeToLive() : config.authnio.manifestCache.ttl;
+
+  // update cache
+  freshUntil = Date.now() + config.authnio.manifestCache.ttl;
+  MANIFEST_CACHE.set(manifestUrl, {policy, response, freshUntil}, ttl);
+
+  // update response header using latest policy
+  response.headers = policy.responseHeaders();
+  return response;
+}
+
+async function getManifest({manifestUrl, request}) {
+  // fetch web app manifest from server
+  const {headers} = request;
+  let response;
+  try {
+    // use a short timeout to help prevent DoS
+    response = await axios.get(manifestUrl, {
+      headers,
+      timeout: config.authnio.manifestCache.requestTimeout,
+      transformResponse(body) {
+        // do not parse JSON
+        return body;
+      }
+    });
+  } catch(e) {
+    if(e.response) {
+      response = e.response;
+    } else {
+      // transform to a 404
+      response = {
+        status: 404,
+        headers: {},
+        data: null
+      };
+    }
+  }
+
+  if(response.status >= 400) {
+    // do not store any response body for errors
+    response.headers = {};
+    response.data = null;
+  }
+
+  // TODO: may need to do something similar for other status codes as well
+  if(response.status === 200 && !_isContentJson(response)) {
+    // transform to 404 to avoid storing spurious response
+    response = {
+      status: 404,
+      headers: {},
+      data: null
+    };
+  }
+
+  // return sanitized response
+  return {
+    status: response.status,
+    headers: response.headers,
+    body: response.data
+  };
+}
+
+function _isContentJson(response) {
+  const contentType = response.headers['content-type'];
+  return (typeof contentType === 'string' &&
+    (contentType.startsWith('application/manifest+json') ||
+      contentType.startsWith('application/json')));
```

Comment:
Could use https://www.npmjs.com/package/content-type to extract type and do exact string matches.  Would handle edge cases like avoiding a bad positive match on 'application/jsonfoobar'.